tests/Integration/Behaviour/Features/Context/Domain/CartRuleFeatureContext.php

```diff
@@ -242,12 +242,14 @@ public function iSaveIt(array $properties)
             $properties['discount_product_id'] ?? null
         );
 
-        ObjectModel::resetStaticCache();
+        ObjectModel::disableCache();
```

Comment:
@matks  I had to disable object model cache because it was conflicting with other scenarios that create cart rules :thinking: another cart rule was loaded out of cache with the same ID.

Do you think this is a good solution? 

I'm also thinking if we should disable the object model cache globally before all test suites, wdyt?

Reply:
The downside I see is it would require to hit database every time resulting in longer builds. But that may not be a problem.

Reply:
> I'm also thinking if we should disable the object model cache globally before all test suites, wdyt?

This is the solution ✅ 

@sarjon If we can use more hardware to deal with it, it's not an issue 😄 but 👍 it's good you thought about it

For any integration tests, we can reset as many mechanisms as needed : database, cache, config, filesystem